boa_engine/src/syntax/parser/expression/primary/mod.rs

```diff
@@ -98,6 +101,9 @@ where
                         .map(Node::from)
                 }
             }
+            TokenKind::Keyword(Keyword::Class) => {
+                ClassExpression::new(self.name, false, false).parse(cursor, interner)
+            }
```

Comment:
I guess the "TODO" a bit higher is the issue. I guess this could be fixed in a future PR.